src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -426,7 +429,7 @@ static CHIP_ERROR GetAddrInfo(void * context, DnssdResolveCallback callback, uin
         err = TXTRecordGetItemAtIndex(txtLen, txtRecord, i, kDnssdKeyMaxSize, key, &valueLen, &valuePtr);
         VerifyOrReturnError(CheckForSuccess(sdCtx, __func__, err, true), CHIP_ERROR_INTERNAL);
 
-        strncpy(value, reinterpret_cast<const char *>(valuePtr), valueLen);
+        Platform::CopyString(value, reinterpret_cast<const char *>(valuePtr));
```

Comment:
I don't think `valuePtr` is null-terminated here.  This should be more like:
```suggestion
        CopyString(value, ByteSpan(static_cast<const uint8_t *>(valuePtr), valueLen));
```
or so.  And probably need to #include Span.h in this file for that.

Reply:
Updated to use memcpy. I thought usage of strncpy and cast to const char * means this was a null terminated string. But if this is just memory, memcpy seems apropriate and no casts needed.